### PR TITLE
Migrate to use swift-html

### PR DESCRIPTION
**Pull request by stephencelis**
We've been on swift-web, so it's time to dogfood the slimmed down lib.

### CODE REVIEW

**Comment by stephencelis** [comment]
(This is still a WIP and has some broken things for us to discuss next week.)

File: Tests/PointFreeTests/__Snapshots__/SiteMiddlewareTests/testWithHttps.localhost_allowed.Conn.txt [hunk 1]
```diff
@@ -0,0 +1,2821 @@
...
+                                      bold
+                                      ts-d-r4
+                                      ts-m-r3
+                                      lh-2
+                                      fg-black"
+                               max="100"
+                               min="2"
+                               name="pricing[quantity]"
+                               onblur="javascript:
+this.value
```
Comment (mbrandonw) [inline Tests/PointFreeTests/__Snapshots__/SiteMiddlewareTests/testWithHttps.localhost_allowed.Conn.txt]: this is funny, we should double check what the pretty printer is doing with this attribute...

File: Sources/PointFree/GoogleAnalytics.swift [hunk 2]
```diff
@@ -7,15 +7,15 @@ public func addGoogleAnalytics(_ nodes: [Node]) -> [Node] {
...
-        return .element(element |> \.content %~ map(addGoogleAnalytics))
-      }
-    case .text:
+    case let .element(tag, attribs, children):
+      return .element(
+        tag,
+        attribs,
+        tag == "head"
+          ? children + [googleAnalytics]
+          : addGoogleAnalytics(children)
```
Comment (mbrandonw) [inline Sources/PointFree/GoogleAnalytics.swift]: much simpler!

File: Sources/PointFree/Routers/PointFreeRoutes.swift [hunk 3]
```diff
@@ -264,10 +264,10 @@ public func url(to route: Route) -> String {
   return router.url(for: route, base: Current.envVars.baseUrl)?.absoluteString ?? ""
 }
 
-extension PartialIso where A == String, B == Tag {
-  public static var tag: PartialIso<String, Tag> {
-    return PartialIso<String, Tag>(
-      apply: Tag.init(slug:),
+extension PartialIso where A == String, B == SiteTag {
```
Comment (mbrandonw) [inline Sources/PointFree/Routers/PointFreeRoutes.swift]: I should follow up and delete `SiteTag`, we never even used it.

File: Sources/PointFree/VideoJs.swift [hunk 4]
```diff
@@ -77,7 +79,16 @@ struct VideoJsOptions: Encodable {
   static let `default` = VideoJsOptions(control: true, playbackRates: [0.5, 1, 1.5, 2])
 
   var jsonString: String {
-    return ((try? String(data: JSONEncoder().encode(VideoJsOptions.default), encoding: .utf8)) ?? nil)
-      ?? "{}"
+    if #available(OSX 10.13, *) {
```
Comment (mbrandonw) [inline Sources/PointFree/VideoJs.swift]: nice catch

File: Sources/PointFree/VideoJs.swift [hunk 5]
```diff
@@ -77,7 +79,16 @@ struct VideoJsOptions: Encodable {
...
+    } else {
+      return ((try? String(data: JSONEncoder().encode(VideoJsOptions.default), encoding: .utf8)) ?? nil)
+        ?? "{}"
+    }
   }
 }
+
+@available(OSX 10.13, *)
+private let jsonEncoder = JSONEncoder()
+  |> \.outputFormatting .~ .sortedKeys
```
Comment (mbrandonw) [inline Sources/PointFree/VideoJs.swift]: I think this is not available in linux right?

Comment (mbrandonw) [reply Sources/PointFree/VideoJs.swift]: now* available in linux

**Comment by mbrandonw** [comment]
> @mbrandonw Think this is _finally_ good to go!

looks like we gotta guard against some macOS imports:

```/home/travis/build/pointfreeco/pointfreeco/Sources/PointFreeTestSupport/PointFreeTestSupport.swift:466:8: error: no such module 'Cocoa'
import Cocoa
       ^
/home/travis/build/pointfreeco/pointfreeco/Sources/PointFreeTestSupport/PointFreeTestSupport.swift:466:8: error: no such module 'Cocoa'
import Cocoa
       ^
error: terminated(1): /home/travis/.swiftenv/versions/4.2/usr/bin/swift-build-tool -f /home/travis/build/pointfreeco/pointfreeco/.build/debug.yaml test output:
    
make: *** [test-oss] Error 1
```

File: Sources/PointFree/VideoJs.swift (hunk 5 above)
Comment (mbrandonw) [reply Sources/PointFree/VideoJs.swift]: oh I see, cool

File: Tests/PointFreeTests/EmailTests/__Snapshots__/NewEpisodeEmailTests/testNewEpisodeEmail_Subscriber.2.txt [hunk 6]
```diff
@@ -13,11 +13,24 @@ Watch now! <http://localhost:8080/episodes/ep2-proof-in-functions>
...
 
+
+
+
+
+
+
+
+
+
```
Comment (mbrandonw) [inline Tests/PointFreeTests/EmailTests/__Snapshots__/NewEpisodeEmailTests/testNewEpisodeEmail_Subscriber.2.txt]: oof just noticed these. I may have translated the plain text render incorrectly. def not a blocker for merging but something to fix later.

File: Tests/PointFreeTests/EmailTests/__Snapshots__/InviteEmailTests/testInviteAcceptance.1.html [hunk 7]
```diff
@@ -1,72 +1,192 @@
...
               </td>
             </tr>
           </table>
-          <table border="0" cellpadding="0" cellspacing="0" class="bg-gray900" style="background-color:#f6f6f6;padding-top:16px;padding-right:16px;padding-bottom:16px;padding-left:16px;max-width:600px;margin-top:0;margin-bottom:0;margin-right:auto;margin-left:auto;display:block">
+          <table border="0"
+                 cellpadding="0"
+                 cellspacing="0"
+                 class="bg-gray900"
+                 style="background-color:#f6f6f6;
+                        padding-top:16px;
```
Comment (mbrandonw) [inline Tests/PointFreeTests/EmailTests/__Snapshots__/InviteEmailTests/testInviteAcceptance.1.html]: breaking out styles on newlines makes reviewing email html so much better!

File: Tests/PointFreeTests/__Snapshots__/BlogTests/testBlogAtomFeed.1.Conn.txt [hunk 8]
```diff
@@ -40,28 +39,27 @@ X-XSS-Protection: 1; mode=block
...
 should be broken out into separate content blocks so that we can use the JS syntax highlighting
 library. For example, here is some code:</p>
-</div></div><pre><code class="code block m-pl3 m-pr3 m-pt3 m-pb3 overflow-auto-x language-swift">struct PredicateSet&lt;A&gt; {
-  let contains: (A) -&gt; Bool
+</div></div><pre><code class="code block m-pl3 m-pr3 m-pt3 m-pb3 overflow-auto-x language-swift">struct PredicateSet&lt;A> {
+  let contains: (A) -> Bool
```
Comment (mbrandonw) [inline Tests/PointFreeTests/__Snapshots__/BlogTests/testBlogAtomFeed.1.Conn.txt]: just wanna confirm these `>` shouldn't be escaped anymore?

Comment (stephencelis) [reply Tests/PointFreeTests/__Snapshots__/BlogTests/testBlogAtomFeed.1.Conn.txt]: Yeah it's safe.

File: Tests/PointFreeTests/EmailTests/__Snapshots__/NewEpisodeEmailTests/testNewEpisodeEmail_Subscriber.2.txt (hunk 6 above)
Comment (stephencelis) [reply Tests/PointFreeTests/EmailTests/__Snapshots__/NewEpisodeEmailTests/testNewEpisodeEmail_Subscriber.2.txt]: Yeah I noticed this. The existing one isn't perfect either, so we just need to figure it out. Opened: https://github.com/pointfreeco/pointfreeco/issues/328

**Pull request merged**